lib/src/web_socket/common/w_socket.dart

```diff
@@ -149,10 +149,10 @@ abstract class CommonWSocket extends Stream implements WSocket {
   @override
   StreamSubscription listen(void onData(event),
       {Function onError, void onDone(), bool cancelOnError}) {
+    // ignore: cancel_subscriptions
     var sub = _incoming.stream
         .listen(onData, onError: onError, cancelOnError: cancelOnError);
     _incomingSubscription = new WSocketSubscription(sub, onDone, onCancel: () {
-      sub.cancel();
```

Comment:
So this always perplexed me, but I'm not able to recreate the expected behavior for streams if I cancel this proxied subscription here. The incoming stream _should_ be cleaned up, but this is probably worth some follow-up investigation at a later date.